packages/account/src/Sections/Security/ClosingAccount/closing-account-reason-form.tsx

```diff
@@ -0,0 +1,266 @@
+import React from 'react';
+import { Field, Form, Formik, FormikErrors, FieldProps, FormikValues } from 'formik';
+import { Checkbox, FormSubmitButton, Input, Text } from '@deriv/components';
+import { localize, Localize } from '@deriv/translations';
+import getCloseAccountReasonsList from 'Constants/account_closing_reasons_list';
+import { TClosingAccountFormValues } from 'Types';
+
+type TClosingAccountReasonFormProps = {
+    onBackClick: () => void;
+    onConfirmClick: (reason: string) => void;
+};
+const initial_form_values: TClosingAccountFormValues = {
+    'financial-priorities': false,
+    'stop-trading': false,
+    'not-interested': false,
+    'another-website': false,
+    'not-user-friendly': false,
+    'difficult-transactions': false,
+    'lack-of-features': false,
+    'unsatisfactory-service': false,
+    'other-reasons': false,
+    other_trading_platforms: '',
+    do_to_improve: '',
+};
+const CHARACTER_LIMIT = 110;
+
+type TFormError = FormikErrors<TClosingAccountFormValues> & {
+    empty_reason?: string;
+    characters_limits?: string;
+};
+
+type TCustomState = {
+    is_checkbox_disabled: boolean;
+    total_checkbox_checked: number;
+    remaining_characters: number;
+    total_accumulated_characters: number;
+};
+type TAction =
+    | { type: 'SET_CHECKBOX_DISABLED'; payload: boolean }
+    | { type: 'SET_TOTAL_CHECKBOX_CHECKED'; payload: number }
+    | { type: 'SET_REMAINING_CHARACTERS'; payload: number }
+    | { type: 'SET_TOTAL_ACCUMULATED_CHARACTERS'; payload: number };
```

Comment:
```suggestion
type TActionType = 'SET_CHECKBOX_DISABLED' | 'SET_TOTAL_CHECKBOX_CHECKED' | 'SET_REMAINING_CHARACTERS' | 'SET_TOTAL_ACCUMULATED_CHARACTERS';
type TAction = { type: TActionType; payload: boolean }
```

Reply:
only SET_CHECKBOX_DISABLED is of type boolean, others are of type number

Reply:
Oops missed that, 
Just a suggestion, we can use write like this then, you can ignore this as well 🙏 
```
type TActionType =
    | 'SET_CHECKBOX_DISABLED'
    | 'SET_TOTAL_CHECKBOX_CHECKED'
    | 'SET_REMAINING_CHARACTERS'
    | 'SET_TOTAL_ACCUMULATED_CHARACTERS';

type TAction =
    | { type: Exclude<TActionType, 'SET_CHECKBOX_DISABLED'>; payload: number }
    | {
          type: Extract<TActionType, 'SET_CHECKBOX_DISABLED'>;
          payload: boolean;
      };

```

Reply:
i think the current implementation is more readable.